indico/modules/events/papers/operations.py

```diff
@@ -238,3 +238,16 @@ def update_reviewing_roles(event, users, contributions, role, assign):
                   'Papers unassigned ({})'.format(orig_string(role.title)), session.user, data=log_data)
     db.session.flush()
     logger.info('Paper reviewing roles in event %r updated by %r', event, session.user)
+
+
+def set_deadline(event, deadline, role, enforce):
+    paper_reviewing_settings.set(event, '{}_deadline'.format(role), deadline)
+    paper_reviewing_settings.set(event, 'enforce_{}_deadline'.format(role), enforce)
+    log_data = {'Enforced': enforce}
+    if deadline:
+        log_data['Deadline'] = deadline.isoformat()
+    else:
+        log_data['Deadline'] = 'None'
+    logger.info("Deadline for paper reviewing role '{}' in event %r set by %r".format(role), event, session.user)
```

Comment:
`'Paper reviewing deadline (%s) set in %r by %r'` - this goes into a logfile, so being concise is perfectly fine. Also, don't use `.format()` for logging!